src/base/utils/foreignapps.cpp

```diff
@@ -132,44 +136,34 @@ namespace
         DWORD type = 0;
         DWORD cbData = 0;
         LPWSTR lpValueName = NULL;
+        std::wstring lpValueNameBuffer;
         if (!name.isEmpty())
         {
-            lpValueName = new WCHAR[name.size() + 1];
+            lpValueNameBuffer.resize(name.size() + 5);
+            lpValueName = lpValueNameBuffer.data();
             name.toWCharArray(lpValueName);
-            lpValueName[name.size()] = 0;
         }
 
         // Discover the size of the value
         ::RegQueryValueExW(handle, lpValueName, NULL, &type, NULL, &cbData);
         DWORD cBuffer = (cbData / sizeof(WCHAR)) + 1;
-        LPWSTR lpData = new WCHAR[cBuffer];
-        LONG res = ::RegQueryValueExW(handle, lpValueName, NULL, &type, (LPBYTE)lpData, &cbData);
-        if (lpValueName)
-            delete[] lpValueName;
+        std::wstring lpDataBuffer(cBuffer + 5, 0);
+        LONG res = ::RegQueryValueExW(handle, lpValueName, NULL, &type, (LPBYTE)lpDataBuffer.data(), &cbData);
+        lpValueNameBuffer.clear();
```

Comment:
It seems to be redundant.

Reply:
> It seems to be redundant.

Maybe not ? The first call is to get the size of the buffer and the second call is to get the data :)
But `lpValueName.clear()` is not used for sure